clients/src/main/java/org/apache/kafka/common/Cluster.java

```diff
@@ -42,17 +43,32 @@
      * Create a new cluster with the given nodes and partitions
      * @param nodes The nodes in the cluster
      * @param partitions Information about a subset of the topic-partitions this cluster hosts
+     * @deprecated Use the Cluster constructor with 4 parameters
      */
+    @Deprecated
     public Cluster(Collection<Node> nodes,
                    Collection<PartitionInfo> partitions,
                    Set<String> unauthorizedTopics) {
-        this(false, nodes, partitions, unauthorizedTopics);
+        this(false, nodes, partitions, unauthorizedTopics, Collections.<String>emptySet());
```

Comment:
Seems like the bigger issue is that using an empty set will change the semantics if they switch to a newer version of the clients lib, won't it? If you use the `INTERNAL_TOPICS` constant, the worst case is that a new one is added, which should be expected to happen sometimes. But using an empty set could easily cause at least tests using things like `MockProducer` to start failing since other code using `Cluster` will now see an empty set instead of the existing set of topics.